src/user.ts

```diff
@@ -211,20 +234,32 @@ export class User extends AsyncCreatable<User.Options> {
     super(options);
     this.org = options.org;
   }
+
   /**
    * Generate default password for a user. Returns An encrypted buffer containing a utf8 encoded password.
    */
-  public static generatePasswordUtf8(): SecureBuffer<void> {
-    // one character of each of the 4 types followed by the remaining random characters, then quasi-randomize
-    const ALLCHARS = UPPER.concat(LOWER, NUMBERS, SYMBOLS.join(''));
-    const password = [UPPER[rand(UPPER)], LOWER[rand(LOWER)], NUMBERS[rand(NUMBERS)], SYMBOLS[rand(SYMBOLS)]]
-      .concat(
-        Array(PASSWORD_LENGTH - 4)
-          .fill('0')
-          .map(() => ALLCHARS[rand(ALLCHARS)])
-      )
-      .sort(() => Math.random() - 0.5);
+  public static generatePasswordUtf8(
+    passwordCondition: PasswordConditions = { length: 13, complexity: 5 }
+  ): SecureBuffer<void> {
+    if (!PASSWORD_COMPLEXITY[passwordCondition.complexity]) {
+      throw SfdxError.create('@salesforce/core', 'user', 'complexityOutOfBound');
+    }
+    if (passwordCondition.length < 8 || passwordCondition.length > 1000)
```

Comment:
NIT: feel free to ignore this, but can we add {} for the second if statement? can we also add a new line between the error handling/throwing section, and the `let password`?